WordPress/Classes/System/WordPressAppDelegate.m

```diff
@@ -518,8 +549,63 @@ - (void)showWelcomeScreenIfNeededAnimated:(BOOL)animated
     }
 }
 
+
+- (void)showSigninScreenIfNeededAnimated:(BOOL)animated
+{
+    if (self.isWelcomeScreenVisible || !([self noSelfHostedBlogs] && [self noWordPressDotComAccount])) {
+        return;
+    }
+
+    UIViewController *presenter = self.window.rootViewController;
+    // Check if the presentedVC is UIAlertController because in iPad we show a Sign-out button in UIActionSheet
+    // and it's not dismissed before the check and `dismissViewControllerAnimated` does not work for it
+    if (presenter.presentedViewController && ![presenter.presentedViewController isKindOfClass:[UIAlertController class]]) {
+        [presenter dismissViewControllerAnimated:animated completion:^{
+            [self showSigninScreenAnimated:animated thenEditor:NO];
+        }];
+    } else {
+        [self showSigninScreenAnimated:animated thenEditor:NO];
+    }
+}
+
+- (void)showSigninScreenAnimated:(BOOL)animated thenEditor:(BOOL)thenEditor
+{
+    [self showSigninScreen:nil animated:animated thenEditor:thenEditor];
+}
+
+- (void)showSigninScreen:(NSString *)token animated:(BOOL)animated thenEditor:(BOOL)thenEditor
+{
+    // Complete a magic link signin if needed.
+    NSString *savedEmail = [SigninHelpers getEmailAddressForTokenAuth];
+    if (token && savedEmail) {
+        SigninLinkAuthViewController *controller = [SigninLinkAuthViewController controller:savedEmail token:token];
+        UINavigationController *navigationController = [[NUXNavigationController alloc] initWithRootViewController:controller];
+        UIViewController *rootViewController = self.window.rootViewController;
+        if (rootViewController.presentedViewController)  {
+            [rootViewController.presentedViewController dismissViewControllerAnimated:NO completion:^{
```

Comment:
> Why do we need dismissing here?

If a UIViewController already has a presentedViewController it will not present another before the first is dismissed.   Its possible when the auth link is opened for the app is resumed vs launched in which case one of the other signin controllers would still be presented. 

Good thinking about the other uses cases. I can tweak the logic here.